### ISSUE TITLE
No tray icon?

### ISSUE THREAD

**Issue by terzag** [BOSSoNe0013/track-o-bot#21]
I compiled Track-o-Bot two monthes ago on Ubuntu with no problem. Tonight, seeing that there was a new version since then, I compiled it again. In the new version, I have no visible tray icon, though the software runs and I can click where the icon should be to access its menu. Any idea why the icon is not displaying/invisible?

Also, there's no more a "quit" option in the menu?

**Comment by BOSSoNe0013** [comment]
Hi,
can you tell me what is your desktop manager (plasma, kde4, gnome, ...) ? I've removed the quit entry as it's display twice on plasma desktop.
Thanks for reporting

**Comment by terzag** [comment]
Using Gnome 3 (3.14).

**Comment by terzag** [comment]
Might be useful: on LXDE, the tray icon displays fine in the panel. I know that Gnome handles the tray in a strange way but it worked fined with Track-o-Bot 0.5.1. Maybe a problem with the size of the icon, like the Gnome panel looking for one at a specific size (16x16? 20x20? 22x22? 24x24?) while other panels adapt.

**Comment by BOSSoNe0013** [comment]
I think icon problem is solved with latest commit

**Comment by terzag** [comment]
Just tried on Lxde: the icon is there (it was there before) and there's still no "quit" option in the menu. Is it supposed to be fixed as well or only the icon problem? I'll check on Gnome for the icon.

But there is a big issue: the new version gives me a different (empty) profile! How do I get my previous one back?

**Comment by BOSSoNe0013** [comment]
Ok done for the quit menu entry.
You can restore your account with Import in Account tab or by editing ~/.config/spidy.ch/Track-o-Bot.conf

**Comment by terzag** [comment]
The quit option is indeed back, thanks.

As for the wrong account, I don't get it: in the Track-o-Bot.conf, I can see the right account, not the one in the settings UI (or the open profile link). As for the import option, I need to have a backup that I exported first before, right?

**Comment by BOSSoNe0013** [comment]
It could be ~/.config/Unknown\ Organization/Track-o-Bot.conf

**Comment by terzag** [comment]
Ok, it's there. I guess the new dir is definitive and I can remove the spidy.ch one?

**Comment by BOSSoNe0013** [comment]
In fact spidy.ch should be the new one, look at the config files content (it's a text file). The one with your disered account should be the old one.

**Comment by terzag** [comment]
Well, that's what I did: the one in Unknown Organization has the new (blank) account while the one in spidy.ch has the good account. It's only after I copied the files from spidy.ch to Unknown Organization that I got the right account back in Track-o-Bot (settings and profile link).

**Comment by angrylinuxuser** [comment]
Missing one line in Trackobot.cpp is at fault here. This fixes a problem with wrong configuration diectory. (trackobot.cpp line 33)

Should be

Trackobot::Trackobot( int argc, char **argv )
  : mApp( argc, argv ),
    mCore( NULL ),
    mWindow( NULL ),
    mSingleInstanceServer( NULL )
  {
    SetupApplication();
  }

but we have empty constructor instead.

**Comment by BOSSoNe0013** [comment]
You're right, i'll update sources ASAP. Thanks

**Comment by BOSSoNe0013** [comment]
@terzag is your problem solved ?

**Comment by terzag** [comment]
The wrong account issue is solved.

The missing quit option issue is solved.

I'm not sure about the tray icon issue as I ditched Gnome 3 and finally settled on Mate (Gnome 2 fork) for my DE. The icon is there but last time I checked, it didn't appear in Gnome 3 and Xfce. I'll try to check again in other DEs.

**Comment by terzag** [comment]
Just checked a bunch of DEs in virtual machines and can confirm the tray icon appears as expected in Lxde, Unity (Ubuntu), Gnome 3, Xfce and Mate. Haven't checked KDE, Cinnamon and elementaryOS (can't think of another DE).

Not sure of the exact version of each but they were from Ubuntu 15.10 variants. Looks like Gnome 3.16 fixed the issue with systray icons and the Track-o-Bot one not appearing in previous version since the problem has been fixed here was a Gnome bug.

Not sure either why it didn't appear last time I checked in Xfce. Either an issue specific to my setup or a bug in xfcd4-panel in previous Xfce version.
